src/Items/ItemBow.h

```diff
@@ -69,17 +69,12 @@ class cItemBowHandler :
 		}
 
 		// Create the arrow entity:
-		cArrowEntity * Arrow = new cArrowEntity(*a_Player, Force * 2);
-		if (Arrow == nullptr)
+		auto ArrowPtr = cpp14::make_unique<cArrowEntity>(*a_Player, Force * 2);
+		auto Arrow = ArrowPtr.get();
+		if (!Arrow->Initialize(std::move(ArrowPtr), *a_Player->GetWorld()))
```

Comment:
> but then I realized, what would get called first, the std::move or the indirection?

OK. I'm going to try to do a Tycho and quote from the standard here.

-----------
From **N4140**:

§ 5.2.2.8
> [Note: The evaluations of the postﬁx expression and of the arguments are all unsequenced relative to one another. All side eﬀects of argument evaluations are sequenced before the function is entered (see 1.9). —end note]

where by postfix expression they mean:

§ 5.2.2.1
> A function call is a postﬁx expression followed by parentheses containing a possibly empty, comma-separated list of initializer-clauses which constitute the arguments to the function. The postﬁx expression shall have function type or pointer to function type.

I'm interpreting that as meaning `->` and `move` do not have a specified order. Initialize really shouldn't be taking owning pointers to itself.